Services/Simulations.cs

```diff
@@ -84,15 +92,19 @@ public class Simulations : ISimulations
             IStorageAdapterClient storageAdapterClient,
             IIotHubConnectionStringManager connectionStringManager,
             IDevices devices,
+            IFileSystem fileSystem,
             ILogger logger,
             IDiagnosticsLogger diagnosticsLogger,
             ISimulationStatistics simulationStatistics)
         {
+            this.config = config;
             this.deviceModels = deviceModels;
             this.storageAdapterClient = storageAdapterClient;
+            this.mainStorage = factory.Resolve<IStorageRecords>().Init(config.MainStorage);
```

Comment:
see my other comment: the agent having a direct dependency on the storage is quite strange and inconsistent with the overall design where all the storage calls go via a service (business logic).

Reply:
resolved other comment